pkg/validations/cluster_test.go

```diff
@@ -95,6 +96,47 @@ func TestValidateCertForRegistryMirrorInsecureSkip(t *testing.T) {
 	tt.Expect(validations.ValidateCertForRegistryMirror(tt.clusterSpec, tt.tlsValidator)).To(Succeed())
 }
 
+func TestValidateAuthenticationForRegistryMirrorNoRegistryMirror(t *testing.T) {
+	tt := newTlsTest(t)
+	tt.clusterSpec.Cluster.Spec.RegistryMirrorConfiguration = nil
+
+	tt.Expect(validations.ValidateAuthenticationForRegistryMirror(tt.clusterSpec)).To(Succeed())
+}
+
+func TestValidateAuthenticationForRegistryMirrorNoAuth(t *testing.T) {
+	tt := newTlsTest(t)
+	tt.clusterSpec.Cluster.Spec.RegistryMirrorConfiguration.Authenticate = false
+
+	tt.Expect(validations.ValidateAuthenticationForRegistryMirror(tt.clusterSpec)).To(Succeed())
+}
+
+func TestValidateAuthenticationForRegistryMirrorAuthInvalid(t *testing.T) {
+	tt := newTlsTest(t)
+	tt.clusterSpec.Cluster.Spec.RegistryMirrorConfiguration.Authenticate = true
+	if err := os.Unsetenv("REGISTRY_USERNAME"); err != nil {
+		t.Fatalf(err.Error())
+	}
+	if err := os.Unsetenv("REGISTRY_PASSWORD"); err != nil {
+		t.Fatalf(err.Error())
+	}
+
+	tt.Expect(validations.ValidateAuthenticationForRegistryMirror(tt.clusterSpec)).To(
+		MatchError(ContainSubstring("please set REGISTRY_USERNAME env var")))
+}
+
+func TestValidateAuthenticationForRegistryMirrorAuthValid(t *testing.T) {
+	tt := newTlsTest(t)
+	tt.clusterSpec.Cluster.Spec.RegistryMirrorConfiguration.Authenticate = true
+	if err := os.Setenv("REGISTRY_USERNAME", "username"); err != nil {
```

Comment:
nit: use `t.Setenv` to set env in tests
```suggestion
	if err := t.Setenv("REGISTRY_USERNAME", "username"); err != nil {
```